pandas/tests/reshape/test_pivot.py

```diff
@@ -2664,46 +2662,46 @@ def test_pivot_columns_not_given(self):
         with pytest.raises(TypeError, match="missing 1 required keyword-only argument"):
             df.pivot()
 
-    @pytest.mark.xfail(
-        using_string_dtype(), reason="TODO(infer_string) None is cast to NaN"
-    )
     def test_pivot_columns_is_none(self):
         # GH#48293
-        df = DataFrame({None: [1], "b": 2, "c": 3})
+        df = DataFrame([[1, 2, 3]], columns=Index([None, "b", "c"], dtype="object"))
```

Comment:
I think we do need to figure out how this should work with the new string data type; the larger discussion to that is being had in https://github.com/pandas-dev/pandas/pull/60329#issuecomment-2480127331